tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/PublicAssetSetMethod.scala

```diff
@@ -0,0 +1,121 @@
+          .fold(SMono.error(_), creationRequest => createPublicAssets(mailboxSession, creationRequest, request.accountId)
+            .map(publicAsset => PublicAssetCreationSuccess(publicAssetCreationId, PublicAssetCreationResponse.from(publicAsset))))
+          .onErrorResume(error => handleError(publicAssetCreationId, error))
+      }.collectSeq()
+      .map(PublicAssetCreationResults)
+
+  private def handleError(publicAssetCreationId: PublicAssetCreationId, error: Throwable): SMono[PublicAssetCreationResult] =
+    error match {
+      case e: PublicAssetException => SMono.just(PublicAssetCreationFailure(publicAssetCreationId, e))
+      case e => throw e
+    }
+
+  private def createPublicAssets(mailboxSession: MailboxSession,
+                                 creationRequest: PublicAssetCreationRequest,
+                                 accountId: AccountId): SMono[PublicAsset] =
+    generatePublicAsset(creationRequest, mailboxSession, accountId)
+      .flatMap(publicAsset => SMono(publicAssetRepository.create(mailboxSession.getUser, publicAsset)))
+
+  private def parseCreate(jsObject: JsObject): Either[PublicAssetCreationParseException, PublicAssetCreationRequest] =
+    PublicAssetCreationRequest.validateProperties(jsObject)
+      .flatMap(validJsObject => PublicAssetSerializer.deserializePublicAssetCreationRequest(validJsObject) match {
+        case JsSuccess(creationRequest, _) => Right(creationRequest)
+        case JsError(errors) => Left(PublicAssetCreationParseException(standardError(errors)))
+      })
+
+  private def generatePublicAsset(creationRequest: PublicAssetCreationRequest,
+                                  session: MailboxSession,
+                                  accountId: AccountId): SMono[PublicAsset] =
+    for {
+      parsedIdentityIds <- parseIdentityIds(creationRequest)
+      uploadId = publicAssetSetService.uploadIdFromBlobId(creationRequest.blobId).fold(e => throw e, identity)
+      identityIds <- publicAssetSetService.checkIdentityIdsExist(parsedIdentityIds, session)
+      uploadMetadata <- getUploadMetadata(session, uploadId)
+      publicAssetId: PublicAssetId = PublicAssetIdFactory.generate()
+      publicURI: PublicURI = publicAssetSetService.generatePublicURI(publicAssetId, accountId)
+    } yield {
+      PublicAsset(
+        id = publicAssetId,
+        publicURI = publicURI,
+        size = uploadMetadata.size,
+        contentType = ImageContentType.from(uploadMetadata.contentType),
+        identityIds = identityIds,
+        blobId = publicAssetSetService.blobIdFromUploadId(uploadId).get)
+    }
+
+  private def getUploadMetadata(session: MailboxSession, uploadId: UploadId): SMono[UploadMetaData] =
+    SFlux(uploadRepository.listUploads(session.getUser)) // it should be better if uploadRepository support method to get upload by uploadId
```

Comment:
Then why not implement it?

Reply:
implemented it in [james-project](https://github.com/linagora/tmail-backend/pull/1057#issuecomment-2123777510) pr